lib/canvas-sketch.js

```diff
@@ -29,6 +29,38 @@ function getTimeProp (oldManager, newSettings) {
   return newSettings.animate ? { time: oldManager.props.time } : undefined;
 }
 
+/**
+ * The main function of the canvasSketch library, responsible for initializing a sketch.
+ *
+ * @param {Function} sketch - The sketch function. This function should be asynchronous and 
+ *                            receive an object with the properties 'width' and 'height'.
+ * @param {Object} settings - The settings object for the sketch.
+ * @param {Array|String} settings.dimensions - The dimensions of your sketch in [width, height] units. If not specified, the sketch will fill the browser window. You can also specify a preset string such as "A4" or "Letter".
```

Comment:
A little extra safety around the array.

```suggestion
 * @param {Array<number>|String} settings.dimensions - The dimensions of your sketch in [width, height] units. If not specified, the sketch will fill the browser window. You can also specify a preset string such as "A4" or "Letter".
```